packages/datagateway-common/src/detailsPanels/dls/visitDetailsPanel.component.test.tsx

```diff
@@ -47,25 +62,26 @@ describe('Visit details panel component', () => {
       endDate: '2019-06-11',
     };
 
-    (useInvestigationDetails as jest.Mock).mockReturnValue({
-      data: rowData,
-    });
-    (useInvestigationSize as jest.Mock).mockReturnValue({
-      data: 1,
-      refetch: jest.fn(),
+    axios.get = jest.fn().mockImplementation((url: string) => {
+      if (/.*\/investigations/.test(url))
+        return Promise.resolve({ data: [rowData] });
+
+      if (/.*\/user\/getSize/.test(url)) return Promise.resolve({ data: 64 });
+
+      return Promise.resolve();
```

Comment:
I'm personally not convinced these mock implementations _need_ to change. The advantages of the original method were:
1. we can make sure we are stubbing the correct method, without having to resort to regex to ensure the right endpoints with the right return data. Someone looking at this file for the first time might be confused as to what is being done here when `axios` itself isn't actually being imported or used in the `visitDetailsPanel` component
2. by mocking the functions themselves, we don't need to care about the _implementation_ of those functions. As far as `visitDetailsPanel` is concerned, it's just calling a function that returns some data and `axios` isn't involved at all. If we did end up changing the implementation of `useInvestigationDetails` or `useInvestigationSize`, we may also have to change the implementation of their mocks here
3. with the new mocking method, we're involving the inner workings of `useInvestigationDetails` and `useInvestigationSize` too much when we really just want to test `visitDetailsPanel` as a unit. `useInvestigationDetails` and `useInvestigationSize` have their own unit tests where we actually do stub `axios` to test them in this fashion. So this isn't really necessary

Reply:
React testing library highly encourages integration testing over unit testing. Since the hooks are used in `VisitDetailsPanel`, I think it is worthwhile to see if the hooks work well with the component itself. If we stub the hooks, it encourages testing the implementation details of the component itself. In the migration to RTL for `datagateway-download`, I used the same approach to test the components as well.

Reply:
Alright I'll go with it, I just think it makes it harder to figure out what's going wrong if something breaks. And what does this mean for the test files in `common/src/api`? Are some of them no longer necessary?

Reply:
The tests in `common/src/api` uses `renderHook` from `testing-library` to test the individual hooks, so I think they are fine. I do agree it is sometimes difficult to figure out what went wrong when things break.